### ISSUE TITLE
(Bug?) ResponseQuery Data changed from Value to Key

### ISSUE THREAD

**Issue by dferendo** [cosmos/ibc-apps#156]
We noticed from testing that the data encoded in the [ResponseQuery](https://github.com/cometbft/cometbft/blob/v0.37.2/proto/tendermint/abci/types.proto#L205) on the host side was changed to be stored into [Key](https://github.com/cometbft/cometbft/blob/v0.37.2/proto/tendermint/abci/types.proto#L211C44-L211C44) instead of [Value](https://github.com/cometbft/cometbft/blob/v0.37.2/proto/tendermint/abci/types.proto#L212) when changing versions from [v7.0.0](https://github.com/cosmos/ibc-apps/releases/tag/modules%2Fasync-icq%2Fv7.0.0) to [v7.1.1](https://github.com/cosmos/ibc-apps/releases/tag/modules%2Fasync-icq%2Fv7.1.1). The tests were ran using both osmosis and juno, unfortunately cannot provide the e2e test since it is in a private repo currently. We can spin up a test on this repo for further proof if required. 

There was no mention of this change in the [CHANGELOG](https://github.com/cosmos/ibc-apps/releases/tag/modules%2Fasync-icq%2Fv7.1.1) and we were wondering whether this change was intentional. Thanks!

**Comment by Reecepbcups** [comment]
@dferendo Thanks for the report. This change was not intentional as I can see.
The only file that was modified with any reference to `abci.ResponseQuery` is:

```
git diff modules/async-icq/v7.0.0 modules/async-icq/v7.1.1 -- modules/async-icq/keeper/relay.go > ~/Desktop/relay.diff
```

A new [`applyFuncIfNoError`](https://github.com/cosmos/ibc-apps/blob/main/modules/async-icq/keeper/cache_ctx.go#L21) was added but nothing in the function indicates to me why it would swap key / value pairings here

Did the function previously panic on v7.0.0?

**Comment by dferendo** [comment]
No, the function was a normal balance query and the data is still correct, just stored in the wrong variable.

**Comment by dferendo** [comment]
@Reecepbcups 

Looking more into it, it seems that [SerializeCosmosResponse](https://github.com/cosmos/ibc-apps/blob/modules/async-icq/v7.1.1/modules/async-icq/keeper/relay.go#L75) is returning different bytes between the two versions. We checked the data in this [unit test](https://github.com/cosmos/ibc-apps/blob/modules/async-icq/v7.1.1/modules/async-icq/keeper/relay_test.go#L33).

v7.0.0 returns 
`eyJkYXRhIjoiQ2dRNkFoSUEifQ==`

v7.1.1 returns 
`eyJkYXRhIjoiQ2dReUFoSUEifQ=`

**Comment by migueldingli1997** [comment]
Turns out this particular unit test is mostly likely returning different bytes because of the address which is different for each test run. That being said we're still experiencing this issue but haven't narrowed it down enough.
